### DISCUSSION TITLE
Change URL of issues link

### DISCUSSION THREAD

**Discussion by Quiwy** [renovatebot/renovate#13492] [Request Help]
Hi,

I would like to change URL of issue link when renovate bot create a new merge request with release note included but I can't find the option.

![image](https://user-images.githubusercontent.com/20048264/148945941-19b76496-dd96-4a61-9cee-63b4639508af.png)
In the image, the green ones are fine and point to my gitlab self hosted instance, but the red ones point to `https://github.com/namespaceOk/projectOk/issueOk`

Is there a way to change this behaviour for issues link to point to `https://my.self.instance/namespaceOk`?

I dit not found something relevant in logs:

![image](https://user-images.githubusercontent.com/20048264/148949986-93541283-b023-4be6-b191-e614dc4a4342.png)

My config file is quite basic :

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "endpoint": "https://my.self.instance",
  "git-submodules": {
    "enabled": true
  },
  "baseBranches": ["develop"],
  "addLabels": ["dependency-update-bot"],
  "branchConcurrentLimit": 2,
  "prConcurrentLimit": 2,
  "prHourlyLimit": 2,
  "dependencyDashboard": true,
  "dependencyDashboardFooter": "Automatically created issue :email: @user.name if needed",
  "packageRules": [
    {
      "schedule": ["every2 weeks on monday"]
    }
  ]
}
```
Not sure of the endpoint parameter, I just added it now to try if it fixes my problem but no.

Any help would be appreciated 😄

**Comment by rarkins** [comment]
I'm not sure I understand what you mean, but I can't think of any way to "rewrite" specific contents of the changelogs, if that's what you're asking.

**Comment by Quiwy** [comment]
Thank you for your answer,

I will try to explain better the problem:

In the release note part created by renovate, there is a link to the source code and to compare with previous version. These links work fine and point to our self hosted gitlab instance.

![image](https://user-images.githubusercontent.com/20048264/149148557-94af1986-914f-4735-b2a1-cafd4b7031f9.png)

On the other hand, the links to the issues directly mentioned does not point to the good domain, neither to the good namespace/project:

![image](https://user-images.githubusercontent.com/20048264/149149055-9f963563-82ec-4c48-8e8b-6dd2653d1452.png)

I would like the link to be `https:/mydomain.com/SW/team/tools/CICD_yaml/-/issues/52` .

The tag release note contains the same text, and the link is fine on gitlab side: 

![image](https://user-images.githubusercontent.com/20048264/149169960-5b75b605-24cf-4d8b-925d-d1a75ccd9108.png)

Is this better understandable?

Furthermore, I just noticed that the release url displayed from renovate is not working either, only comparison url is ,working.

The release url misses a "-". Right now it's  `https://mydomain.com/SW/team/tools/CICD_yaml/releases/v1.4.2`, where it should be `https://mydomain.com/SW/team/tools/CICD_yaml/-/releases/v1.4.2`. See  [this merge request](https://gitlab.com/gitlab-org/gitlab-foss/-/merge_requests/28830) of gitlab for more informations.

I use v 31.23.2 of renovate docker image.

Thank you very much for the help !

**Reply by rarkins** [reply]
So the release notes are hosted on your private gitlab instance, and they include "short" issue links like `foo/bar#52`, but those numbers are being translated into github.com/foo/bar links? Or are the issue links fully raw like `#52`?

**Reply by Quiwy** [reply]
It the first one. The release note includes shorts links like `My text #52` with `#52` being translated directly 

The numbers are being translated into github.com and the last part of the namespa/project is not present.

I tried to reproduce the configuration with 3 examples projects :

- renovate runner : https://gitlab.com/quiwy/renovate-runner/-/tree/main
- project to include : https://gitlab.com/quiwy/lib-1 with release of v2.0.0 : https://gitlab.com/quiwy/lib-1/-/tags/v2.0.0
- project which includes lib-1 : https://gitlab.com/quiwy/example-for-include
- merge request created by renovate : https://gitlab.com/quiwy/example-for-include/-/merge_requests/3

The only difference with the behavior I have on my private instance is that the short link seems to not be a link, don't know why 😞 

I hope with this example it's clearer.

**Reply by rarkins** [reply]
I think we'd need to do quite some massing functionality to get this to work. e.g. search the markdown for all occurrences of `#123`-like issue/PR/MR numbers, and then try to fully specify them based on the platform + repo, e.g. `https://gitlab.me.com/group/repo#123`

**Reply by Quiwy** [reply]
Is there a way to disable the short link being translated to github.com? I would rather like to have no link than to have a bad link.

**Reply by rarkins** [reply]
There is not, but we'd accept a PR if you can find a way to add that capability. I believe this is the function which performs that link transformation: https://github.com/renovatebot/renovate/blob/4f65b57225a42f9deb010f8dbfeb06eb94c0dd69/lib/util/markdown.ts#L30-L40

Here is where it's called from: https://github.com/renovatebot/renovate/blob/4f65b57225a42f9deb010f8dbfeb06eb94c0dd69/lib/workers/pr/changelog/release-notes.ts#L120-L124
